terraform/modules/benchmark/service/benchmark_base.rb

```diff
@@ -25,15 +28,34 @@ def measure_online_validation(benchmark, pid, metrics)
         end
         sleep 0.1
       end
+
+      # Sleep for the specified validation duration
       sleep benchmark.online_validation_secs
+
+      # Calculate the number of epochs
       last_epoch = get_last_epoch(benchmark)
       metrics[:num_epochs] = last_epoch - first_epoch
-
       @logger.info 'Stopping process...'
-      benchmark.stop_command(pid)
+
+      # Identify and stop the child process
```

Comment:
Perhaps this should be a part of another PR. What does it have to do with measuring peak memory usage?

Reply:
I don't quite understand. Do you mean that stopping the process of `/usr/bin/time <some-command>` does not stop its child processes? They are left orphaned?

Reply:
I don't think this is the right way to handle this, then. There are standard UNIX tools to manage killing a parent with their children. Either use the group PID to kill the parent (it should then kill the children as well, as they share this identifier) or alternatively there's a `-P` flag for `pkill` (some logic would need to be changed for this, to my understanding of the script)